external/runtime/runtime.depproj

```diff
@@ -1,7 +1,11 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <RuntimeIdentifier>$(PackageRID)</RuntimeIdentifier>
+    <!--
+      Use the configuration default for PackageRID. PackageRID might be overridden by the build
+      command, and interfere with this depproj in -allconfigurations mode.
+    -->
+    <RuntimeIdentifier>$(PackageRIDConfigurationDefault)</RuntimeIdentifier>
```

Comment:
I think this will result in folks incorrectly building against the local system RID when they are trying to cross-target.

Consider, I'm on win-x64 and trying to build for win-arm by passing in PackageRID.

Reply:
I think the fix should be to derive RuntimeIdentifier from OSGroup instead, but I haven't thought through all the cases.

Reply:
Submitted https://github.com/dotnet/corefx/pull/34409 to revert, we can carry it as a patch in source-build (or submit another PR with a `DotNetBuildFromSource` condition) until there's a better solution.

While working on this change, I was also thinking about if I could make the system use `PackageRID` if the RID "applies" to the current configuration. Seems like that wouldn't have interfered with cross-targeting. I wasn't able to figure it out, though.

I was thinking that in our case, `linux-x64` would be replaced by `centos-x64` when restoring for `netcoreapp-Unix`/`netcoreapp`, but `netcoreapp-Windows_NT` would still get `win-x64`. My idea on how to implement would be something like `/p:PackageRIDAppliesToConfiguration=netcoreapp-Unix`, but I think there's more to applicability than that.

@ericstj, do you know if you'll be able to work on a better fix?

Reply:
I'm re-reading the original issue and I don't understand why you would be specifying PackageRID for -AllConfigurations: can you elaborate on why you need to do this?

Reply:
We need the packages that come out of the `-AllConfigurations` build but also the runtime packages that core-setup uptakes.

Thinking back, I think I remember a really old version of source-build actually running the corefx build twice--maybe that was once to run `-AllConfigurations` and then again for runtime packages? Seems like that approach would fit better with the corefx infra's current expectations, but really not fun for our build times.

Reply:
A few other factors are that we need to support running on non-supported distros (I believe this means we need to pass `PackageRID`, can't rely on default for platform doing the build) and both portable and non-portable builds.

Reply:
> running on non-supported distros

I'd expect this is needed only for non-all-configurations builds, unless you added a new non-supported-distro-asset to some standalone package.  In that case, I'd expect that you only want to build that new non-supported-distro-asset against the new RID's runtime, not all runtime-dependent assets.

Reply:
You may want to check requirements a bit.  Back when we split up the build in 2.0 the whole purpose of the "allConfigurations" leg was to get the things that weren't needed for vertical/source-build out of the main build legs.  All the stuff we build in allConfigurations requires desktop / uap / past packages (what you're calling prebuilts I think) so I would have expected all of that to be out of scope for source-build.

Reply:
Yeah, prebuilts are any packages that we have to download to build. In some cases these are only used as compile-time references--we think we can reasonably decompile those to plaintext, include them in the tarball, and regenerate them during the build. But when they're used by the the product (SDK+Runtime) at runtime, we want them to build from the original sources.

If we were only source-building the Runtime, yeah, one vertical would be enough. However, the SDK brings in stuff like Roslyn which uses implementations from `System.Collections.Immutable` and `System.Reflection.Metadata`. (I think ASP.NET Core will bring in even more, once they're in source-build.) Maybe we can be more specific than `-allconfigurations`, but we do need some way to build these extras.

> All the stuff we build in allConfigurations requires desktop / uap / past packages

Hmm. I was thinking this is fine as long as they're only used as compile-time references, since we have the decompilation plan. But package harvesting from old versions seems like a bigger problem. It seems risky to just skip it, since that would make source-built packages different from Microsoft-built ones. Maybe it's reasonable, though, since our SDK should only be running using the just-built Runtime?

Making the old packages plaintext by using ILDAsm/ILAsm might be a reasonable way to get access to the past packages without any changes. I don't know how round-trippable that is, though, and the IL might be huge.